src/main/scala/com/microsoft/hyperspace/index/sources/default/DefaultFileBasedSource.scala

```diff
@@ -72,8 +73,20 @@ class DefaultFileBasedSource(private val spark: SparkSession) extends FileBasedS
           Hdfs.Properties(Content.fromLeafFiles(files, fileIdTracker).get)
         val fileFormatName = fileFormat.asInstanceOf[DataSourceRegister].shortName
 
+        // Use case-sensitive map if the provided options are case insensitive.
+        val caseSensitiveOptions = options match {
+          case map: CaseInsensitiveMap[String] => map.originalMap
+          case map => map
+        }
+
+        // Get basePath of hive-partitioned data sources, if applicable.
+        val basePathOpt = partitionBasePath(location) match {
```

Comment:
`val basePathOpt = partitionBasePath(location).flatten.map("basePath" -> _)`
Could you use this? instead of creating a Map for this.

Reply:
thanks for the suggestion. this can be done only if we make partitionBasePath a `final` and guarantee that there will never be a None returned from it. This will just introduce bugs in the future if we ever extend DefaultFileBasedSource.

Reply:
@apoorvedave1 There is no hole in this. 

flatten makes 'Option[Option[String]] to Option[String]'

And map only performed the Option is Some and it makes the result Option[(String,String)]

Then Map() ++ Option[(String, String)] added the element only if it's Some.

Reply:
got it. to me the current one is clear enough (adding map to a map vs option to a map). I will let others take a call on this one. Thanks @sezruby

Reply:
Note that the overhead is not the same..